scratch/copy-mk.py

```diff
@@ -0,0 +1,162 @@
+    except (aiokatcp.FailReply, ConnectionError) as exc:
+        print(f"Error: {exc}", file=sys.stderr)
+        return 1
+
+    if len(informs) == 0:
+        print(f"No subarrays currently running on {args.cmc_host}")
+        return 1
+
+    subordinates = {}
+    for inform in informs:
+        name = inform.arguments[0].decode()
+        control_port, monitor_port = inform.arguments[1].decode().split(",")
+        control_port = int(control_port)
+        monitor_port = int(monitor_port)
+        mcast_groups = []
+        for grp in inform.arguments[2:]:
+            mcast_groups.append(grp.decode())
+        n_antennas = len(mcast_groups) // 2
+        for n, grp in enumerate(mcast_groups[1:]):
+            if grp == mcast_groups[0]:
+                # MK is not using a full power-of-2 size array, there are dummies from here onwards.
+                print("Dummy antenna found starting at ", (n + 1) // 2)
+                n_antennas = (n + 1) // 2
+                break
+        digitiser_address = mcast_groups[0].split("+")[0]
+
+        subordinates[name] = Subordinate(name, control_port, monitor_port, digitiser_address, n_antennas)
+
+    try:
+        target = subordinates[args.target]
+    except KeyError:
+        print(f"{args.target} is not among the subordinates currently running. Options are: {subordinates.keys()}")
+        return 1
+
+    # connect to the corr2_servlet itself to see the sync time, n_chans and bandwidth
+    corr2_client = await aiokatcp.Client.connect(args.cmc_host, control_port)
+    sync_time = await get_sensor_value(corr2_client, "sync-time", float)
+    channels = await get_sensor_value(corr2_client, "antenna-channelised-voltage-n-chans", int)
+    bandwidth = await get_sensor_value(corr2_client, "antenna-channelised-voltage-bandwidth", float)
+    # TODO: this could probably be better, and doesn't support S-band yet.
+    match bandwidth:
+        case 544000000.0:
+            band = "u"
+        case 856000000.0:
+            band = "l"
+        case _:
+            print(f"Unable to determine which band we are in. Bandwidth reported as {bandwidth}", file=sys.stderr)
+            return 1
+
+    out_kwargs = {
+        "name": target.name.replace(".", "_"),
+        "antennas": target.n_antennas,
+        "channels": channels,
+        "digitiser-address": mcast_groups[0].split("+")[0],
+        "sync-time": sync_time,
+        "band": band,
+    }
+
+    out_cmd = [
+        "python",
+        "sim_correlator.py",
+        "cbf-mc.cbf.mkat.karoo.kat.ac.za",
+    ]
+
+    for k, v in out_kwargs.items():
+        out_cmd.append(f"--{k}={v}")
+
+    if args.develop is not None:
+        out_cmd.append(f"--develop={args.develop}")
+
+    if args.image_override is not None:
+        for override in args.image_override:
+            out_cmd.append(f"--image-override={override}")
+
+    output = subprocess.run(out_cmd, capture_output=True)
+    print(output)
```

Comment:
Is there a particular reason you want to capture the output, rather than letting the command just write directly to the stdout?

You could also consider using execve (forget the exact Python interface), which will basically replace this process with the out_cmd so that you don't need to worry about trying to proxy anything like output and return codes.